src/core/public/public.api.md

```diff
@@ -544,30 +544,27 @@ export interface HttpRequestInit {
 }
 
 // @public (undocumented)
-export interface HttpServiceBase {
-    addLoadingCount(countSource$: Observable<number>): void;
+export interface HttpSetup {
+    // Warning: (ae-forgotten-export) The symbol "LoadingCountSetup" needs to be exported by the entry point index.d.ts
+    // Warning: (ae-unresolved-link) The @link reference could not be resolved: The package "kibana" does not have an export "LoadingCountSetup"
+    addLoadingCount: LoadingCountSetup['addLoadingCount'];

```

Comment:
NIT: this is why I think we should stop using the `Internal[XXX]` and actually make the 'effort' of adding proper public interfaces for all exported contracts. This is additional boilerplate I agree, but we will need them anyway when we'll cleanup the documentation.

This is a shame our doc generator is not able to properly 'replace' this kind of definition on the fly though.

Reply:
> This is a shame our doc generator is not able to properly 'replace' this kind of definition on the fly though.

I agree. I think part of the complexity of building a tool that does this is knowing when to show the resolved type and when to show a link to a different page.

For now, I'll switch this to put the documented definition on the HttpSetup interface and reference it from the LoadingCountSetup interface.